src/Bicep.Core/Parser/Parser.cs

```diff
@@ -410,6 +449,33 @@ private IdentifierSyntax Identifier(DiagnosticBuilder.ErrorBuilderDelegate error
             return new IdentifierSyntax(identifier);
         }
 
+        private IdentifierSyntaxBase IdentifierWithRecovery(DiagnosticBuilder.ErrorBuilderDelegate errorFunc, params TokenType[] terminatingTypes)
+        {
+            var identifierOrSkipped = this.WithRecovery(
+                () => Identifier(errorFunc),
+                RecoveryFlags.None,
+                terminatingTypes);
+
+            switch (identifierOrSkipped)
+            {
+                case IdentifierSyntax identifier:
+                    return identifier;
+
+                case SkippedTriviaSyntax skipped when skipped.Elements.Any() == false:
+                    var position = this.reader.Peek().Span.Position;
+                    return new MalformedIdentifierSyntax(new[] {CreateMissingToken(position)}, skipped.Diagnostics);
+
+                case SkippedTriviaSyntax skipped:
+                    // WithRecovery only skips tokens, so the cast should be safe
+                    return new MalformedIdentifierSyntax(skipped.Elements.Cast<Token>(), skipped.Diagnostics);
```

Comment:
Any way to do away with the cast? Could we have `WithRecovery` (or an overload) return the tokens rather than `SkippedTriviaSyntax`? Or alternatively, move it into `SkippedTriviaSyntax` itself?

Reply:
The `SkippedTriviaSyntax` refactored removed the cast entirely.